src/config.rs

```diff
@@ -73,17 +83,110 @@ impl Config {
 
         config.api_key = file.api_key;
 
-        if let None = config.lang {
-            if let Some(code) = file.lang {
-                config.lang = code.parse().map_or_else(
-                    |e| {
-                        println!("Config file: {}", e);
-                        None
-                    },
-                    |c| Some(c),
-                )
+        config.lang = if let Some(_) = opt.lang {
+            opt.lang
+        } else if let Some(code) = file.lang {
+            code.parse().map_or_else(
+                |e| {
+                    println!("Config file: {}", e);
+                    None
+                },
+                |c| Some(c),
+            )
+        } else {
+            None
+        };
+
+        config.ascended = if let Some(provided) = opt.ascended_value {
+            if let Some(value) = provided {
+                Some(value)
+            } else {
+                Some(0)
             }
-        }
+        } else if let Some(currencies) = &file.currencies {
+            if let Some(value) = currencies.ascended {
+                Some(value)
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
+        config.karma = if let Some(value) = opt.karma {
+            Rational32::approximate_float(value)
+        } else if let Some(currencies) = &file.currencies {
+            if let Some(value) = currencies.karma {
+                Rational32::approximate_float(value)
+            } else {
+                None
+            }
```

Comment:
This can also be applied to `um`, `vm` and `rn` below:
```suggestion
            currencies.karma.or_else(Rational32::approximate_float)
```